src/test/java/net/gcardone/junidecode/JunidecodeTest.java

```diff
@@ -92,6 +96,11 @@ public void someTests() {
             String decoded = Junidecode.unidecode(sp.getLeft());
             assertEquals(sp.getRight(), decoded);
             sevenBitPurity(decoded);
+            // lower case
+            String lowerCase = Junidecode.unidecode(sp.getLeft(), Case.LOWER_CASE);
+            assertEquals(sp.getRight().toLowerCase(Locale.ROOT), lowerCase);
+            String upperCase = Junidecode.unidecode(sp.getLeft(), Case.UPPER_CASE);
+            assertEquals(sp.getRight().toUpperCase(Locale.ROOT), upperCase);
```

Comment:
This is missing a test to ensure that `null` does not change the case.

Reply:
The existing test code above is calling `Junidecode.unidecode(sp.getLeft())` and testing the result.
And we know this:
```
  public static String unidecode(final String s) {

    return unidecode(s, null);
  }
```

So what are we missing to cover or what needs to be added?

Reply:
FYI:
https://github.com/m-m-m/text/blob/master/ascii/src/main/java/io/github/mmm/text/ascii/impl/package-info.java

BTW: I found and fixed many many mistakes and one bug in Junitdecode. If you are interested I can give some details or your compare your implementation with mine.

Reply:
> No reaction, no progress, no release. Simple PR now hanging for almost a year. Sorry to say so but I will create my own forked library of this code and solve the things I need on my own instead of being blocked so long...

Sorry, life takes over and I consider Junidecode fundamentally feature complete, so new features are low in my priority list. I should have stated this more openly.

To be clear: a fast path to convert ASCII-7 strings to uppercase and lowercase is more generic than Junidecode and there's no need for it to be coupled with `unidecode()`, so I think it's better implemented separately.

Looking at your library it has way bigger aspirations than Junidecode, so forking it seems to me a reasonable thing to do.

> I found and fixed many many mistakes and one bug in Junitdecode. If you are interested I can give some details or your compare your implementation with mine.

Please do open issues if you find bugs or missing codepoints (for example I'm aware that I'm missing the mapping from U+20BF (₿) to `BTC`).